EngineLayer/CommonParameters.cs

```diff
@@ -8,115 +8,146 @@ namespace EngineLayer
-        public bool DoPrecursorDeconvolution { get; set; }
-        public bool UseProvidedPrecursorInfo { get; set; }
-        public double DeconvolutionIntensityRatio { get; set; }
-        public int DeconvolutionMaxAssumedChargeState { get; set; }
-        public Tolerance DeconvolutionMassTolerance { get; set; }
-
-        public int TotalPartitions { get; set; }
-
-        public bool BIons { get; set; }
-
-        public bool YIons { get; set; }
-
-        public bool ZdotIons { get; set; }
-
-        public bool CIons { get; set; }
-
-        public Tolerance ProductMassTolerance { get; set; }
-        public Tolerance PrecursorMassTolerance { get; set; }
-
-        public bool ConserveMemory { get; set; }
-
-        public double ScoreCutoff { get; set; }
-
-        public DigestionParams DigestionParams { get; set; }
-
-        public bool ReportAllAmbiguity { get; set; }
-
-        public int TopNpeaks { get; set; }
-        public double MinRatio { get; set; }
-        public bool TrimMs1Peaks { get; set; }
-        public bool TrimMsMsPeaks { get; set; }
         public string TaskDescriptor { get; set; }
-
-        public bool UseDeltaScore { get; set; }
-        public bool CalculateEValue { get; set; }
+        public int MaxThreadsToUsePerFile { get; private set; }
+        public IEnumerable<(string, string)> ListOfModsFixed { get; private set; }
+        public IEnumerable<(string, string)> ListOfModsVariable { get; private set; }
+        public bool DoPrecursorDeconvolution { get; private set; }
+        public bool UseProvidedPrecursorInfo { get; private set; }
+        public double DeconvolutionIntensityRatio { get; private set; }
+        public int DeconvolutionMaxAssumedChargeState { get; private set; }
+        public Tolerance DeconvolutionMassTolerance { get; private set; }
+        public int TotalPartitions { get; private set; }
+        public bool BIons { get; private set; }
+        public bool YIons { get; private set; }
+        public bool ZdotIons { get; private set; }
+        public bool CIons { get; private set; }
+        public Tolerance ProductMassTolerance { get; private set; }
+        public Tolerance PrecursorMassTolerance { get; private set; }
+        public bool CompIons { get; private set; }
+        public double ScoreCutoff { get; private set; }
+        public DigestionParams DigestionParams { get; private set; }
+        public bool ReportAllAmbiguity { get; private set; }
+        public int TopNpeaks { get; private set; }
+        public double MinRatio { get; private set; }
+        public bool TrimMs1Peaks { get; private set; }
+        public bool TrimMsMsPeaks { get; private set; }
+        public bool UseDeltaScore { get; private set; }
+        public bool CalculateEValue { get; private set; }
 
         #endregion Public Properties
 
         #region Public Methods
 
         public CommonParameters Clone()
         {
-            return (CommonParameters)this.MemberwiseClone();
```

Comment:
Doesn't the old code accomplish the same thing more succinctly?

Reply:
MemberwiseClone is a "shallow copy", it just duplicates the reference to the object. "Deep clone" (copying the properties of an object to a new object) is missing from C#. A deep copy is preferable in most cases. this revised method gets closer to that but it's still a bit problematic because it's still copying references to the same DigestionParams, etc and not making new ones